app/modules/gpio.c

```diff
@@ -32,8 +33,8 @@ static void gpio_intr_callback_task (task_param_t param, uint8 priority)
     // GPIO callbacks are run in L0 and inlcude the level as a parameter
     lua_State *L = lua_getstate();
     NODE_DBG("Calling: %08x\n", gpio_cb_ref[pin]);
-    // The trigger is reset before the callback, as the callback itself might reset the trigger
-    pin_trigger[pin] = true;
+    // Reenable the interrupt just before calling the callback
+    platform_gpio_intr_init(pin, pin_int_type[pin]);
```

Comment:
I was conflicted about this too. We could always do "enable before for edges", "enable after for level" -- one of my versions had that logic, but I felt it was over complex. However, given your comments, I suspect that it might ahve been the right thing to do after all.